### ISSUE TITLE
MockProvider returning undefined when client is declareted

### ISSUE THREAD

**Issue by guhcostan** [apollographql/apollo-client#8732]
**Intended outcome:**
I'm trying to mock some queries I have in my app, but adding client options in useLazyQuerie or mock options to work.

**Actual outcome:**

Screenshot from mock:
![Captura de Tela 2021-08-30 às 18 41 49](https://user-images.githubusercontent.com/22989468/131409498-cfc5450d-0418-401f-8d2c-20124d612920.png)

Screenshot from useLazyQuerie:
![Captura de Tela 2021-08-30 às 18 42 17](https://user-images.githubusercontent.com/22989468/131409564-4dbba1d7-8fbc-40aa-af04-950f779281fb.png)

Screenshot of result with client:
![Captura de Tela 2021-08-30 às 18 37 53](https://user-images.githubusercontent.com/22989468/131409209-ff56859a-55c0-4e69-ae99-bf3c54078278.png)

Screenshot of result without client:
![Captura de Tela 2021-08-30 às 18 38 21](https://user-images.githubusercontent.com/22989468/131409207-bec795ad-2ace-42eb-aa1f-ef670d4de023.png)

**How to reproduce the issue:**
Basically to reproduce the error I created a querie and used useLazyQuerie to fetch the data. When I don't put the client, it works normally, but when I put it, it stops working.

**Versions**
System:
    OS: macOS 11.4
  Binaries:
    Node: 14.15.5 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 6.14.11 - /usr/local/bin/npm
  Browsers:
    Chrome: 92.0.4515.159
    Safari: 14.1.1
  npmPackages:
    @apollo/client: ^3.4.8 => 3.4.10 
    apollo-link-queue: ^3.1.0 => 3.1.0 
    apollo3-cache-persist: ^0.9.1 => 0.9.1 
  npmGlobalPackages:
    apollo: 2.32.5
